drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -198,41 +420,31 @@ static int eth_esp32_dev_init(const struct device *dev)
 	wifi_init_config_t config = WIFI_INIT_CONFIG_DEFAULT();
 	esp_err_t ret = esp_wifi_init(&config);
 
-	ret |= esp_supplicant_init();
+	ret |= esp_wifi_set_mode(WIFI_MODE_STA);
 	ret |= esp_wifi_start();
 
-	if (IS_ENABLED(CONFIG_ESP32_WIFI_STA_AUTO)) {
-		wifi_config_t wifi_config = {
-			.sta = {
-				.ssid = CONFIG_ESP32_WIFI_SSID,
-				.password = CONFIG_ESP32_WIFI_PASSWORD,
-			},
-		};
-
-		ret = esp_wifi_set_mode(WIFI_MODE_STA);
-		ret |= esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config);
-		ret |= esp_wifi_connect();
-	}
-
 	if (ret != ESP_OK) {
-		LOG_ERR("Connect failed");
+		LOG_ERR("Wifi driver failed");
```

Comment:
Same here, posix error codes are expected in general in Zephyr, so please convert those somehow before returning.